waspc/src/Analyzer/TypeChecker/TypeError.hs

```diff
@@ -0,0 +1,37 @@
+module Analyzer.TypeChecker.TypeError
+  ( TypeError (..),
+    TypeCoerceReason (..),
+  )
+where
+
+import Analyzer.Type
+import Analyzer.TypeChecker.AST
+
+data TypeError
+  = UnificationError TypeCoerceReason Type Type
+  | WeakenError TypeCoerceReason TypedExpr Type
+  | NoDeclarationType String
+  | UndefinedIdentifier String
+  | QuoterUnknownTag String
+  | DictDuplicateField String
+  | -- | Temporary "solution" to missing type inference for empty lists
```

Comment:
Put a `TODO:` or a `NOTE:` in front of this comment, just to make it clear something is not as it should be here. Maybe a `TODO:` is best.